src/modules/ekf2/EKF/aid_sources/gnss/gps_checks.cpp

```diff
@@ -136,6 +137,29 @@ void Ekf::collect_gps(const gnssSample &gps)
 
 bool Ekf::runGnssChecks(const gnssSample &gps)
 {
+	// Check GPS spoofing
+	// hysterisis: spoofing needs to be consistent for 1 second before it is declared
+	constexpr uint64_t hysteresis_gps_spoofing_us = 1e6; // 1 second, hardcoded
```

Comment:
1. Does GNSS spoofing really require its own dedicated hysteresis (remember we have that on all these checks as a whole as well)
2. If yes, then let's start using `src/lib/hysteresis/hysteresis.h` for these to keep it simple. https://github.com/PX4/PX4-Autopilot/blob/ac1effa32a4bc4541481778295d4a0d0c213f86f/src/lib/hysteresis/hysteresis.h

Reply:
Using the outside hysteresis would still lead to skipping the GPS-fusion from the very first detection. With the internal one (this), it would still use the GPS data for single false positives.
I don't know which one you prefer. Although, the outside hysteresis has a lag of 7 seconds.

Reply:
I think it would be better to immediately stop the fusion, even if it is a false positive as we would prevent polluting the state if it is a true positive. It is acceptable to dead-reckon for a couple of seconds. The separate hysteresis is probably not required as we monitor `SPOOFING_STATE_MULTIPLE`.
Maybe we can also reduce a bit the default value for the minimum GNSS health time.